Public/Src/Cache/ContentStore/Distributed/Utilities/GrpcDistributedPathTransformer.cs

```diff
@@ -21,7 +21,7 @@ namespace BuildXL.Cache.ContentStore.Distributed.Utilities
     public class GrpcDistributedPathTransformer : IAbsolutePathTransformer
     {
         private readonly IReadOnlyDictionary<AbsolutePath, AbsolutePath> _junctionsByDirectory;
-        private static readonly string _localMachineName = Environment.MachineName;
+        private static readonly string _localMachineName = System.Net.Dns.GetHostName();

```

Comment:
>em.Net.Dns.GetHostName(); [](start = 63, length = 25)

I think this might fail with certain network configuration. We might need some catching & logging around this...

Reply:
This is a good point. My own worry level is fairly low, but if you put the initialization in a static constructor, and throw or log if an exception is thrown, that's probably enough for us to figure out what's happening.

---
In reply to: [306080394](https://github.com/microsoft/BuildXL/pull/645#discussion_r306080394) [](ancestors = 306080394)